src/main/java/org/folio/gobi/Mapper.java

```diff
@@ -823,12 +823,35 @@ private void mapVendorDetail(List<CompletableFuture<?>> futures, VendorDetail ve
       vendorDetail.setReferenceNumbers(referenceNumbers);
     });
 
-    Optional.ofNullable(mappings.get(Mapping.Field.VENDOR_ACCOUNT))
+    Optional.ofNullable(mappings.get(Mapping.Field.VENDOR))
       .ifPresent(field -> futures.add(field.resolve(doc)
-        .thenAccept(o -> vendorDetail.setVendorAccount((String) o))
+        .thenAccept(o -> {
+          if (o != null) {
+          Organization organization = (Organization) o;
+          mapVendorAccount(futures,organization,vendorDetail,doc);
+          }
+        })
         .exceptionally(Mapper::logException)));
   }
 
+  private void mapVendorAccount(List<CompletableFuture<?>> futures, Organization organization, VendorDetail vendorDetail, Document doc)
+  {
+    Optional.ofNullable(mappings.get(Mapping.Field.VENDOR_ACCOUNT))
+      .ifPresent(vendorAccountField -> {
+        String vendorAccountNo = (String) vendorAccountField.resolve(doc).join();
+        Optional.ofNullable(organization.getAccounts().get(0).getAccountNo())
+          .ifPresentOrElse(organizationAccountNo ->{
```

Comment:
Organization can have or can't have account and line "organization.getAccounts().get(0)" in that case will throw NPE